src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj

```diff
@@ -130,7 +130,7 @@ This package is an internal implementation of the .NET Core SDK and is not meant
       <!-- Exclude transitive external dependencies that are not directly referenced by projects in AspNetCore or Extensions -->
       <AspNetCoreReferenceAssemblyPath
           Include="@(ReferencePathWithRefAssemblies)"
-          Condition="'%(ReferencePathWithRefAssemblies.IsReferenceAssembly)' == 'true'"
+          Condition="'%(ReferencePathWithRefAssemblies.IsReferenceAssembly)' != 'false'"
```

Comment:
Have trouble believing the previous `Exclude` item was wrong. Instead the ref/ project shouldn't provide metadata in `@(ReferencePathWithRefAssemblies)`.

Reply:
Ahh, I think I now understand what you mean. We should probably investigate what's happening here.

Reply:
I believe we need to exclude `%(IsReferenceAssemblyProject)` items from `@(AspNetCoreAppReference)` in the main `@(Reference)` item group at https://github.com/dotnet/aspnetcore/blob/0e4635c7562dbe3ce27db27837439c2685bd0336/src/Framework/ref/Microsoft.AspNetCore.App.Ref.csproj#L62 then add another with `<ReferenceOutputAssembly>false</ReferenceOutputAssembly>` for the `@(AspNetCoreAppReference)` items that are from ref/ projects. Might be possible to shorten the new code to
``` msbuild
<ItemGroup>
  <Reference Include="@(AspNetCoreAppReference);@(AspNetCoreAppReferenceAndPackage);@(ExternalAspNetCoreAppReference)">
    <ReferenceOutputAssembly Condition=" '%(IsReferenceAssemblyProject)' == 'true' ">false</ReferenceOutputAssembly>
  </Reference>
```

Reply:
Might be fine to remove references to the ref/ projects entirely i.e. exclude `@(AspNetCoreAppReference)` and `@(AspNetCoreAppReferenceAndPackage)` items with `'%(IsReferenceAssemblyProject)' == 'true'` metadata. Of course, need to confirm that metadata is present. (Root cause here is those two item groups map names to projects and the names are ambiguous.)